de.fu_berlin.inf.dpp.whiteboard/src/de/fu_berlin/inf/dpp/whiteboard/ui/browser/BrowserSXEBridge.java

```diff
@@ -0,0 +1,95 @@
+package de.fu_berlin.inf.dpp.whiteboard.ui.browser;
+
+import org.apache.log4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import de.fu_berlin.inf.dpp.whiteboard.net.WhiteboardManager;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.ISXEMessageHandler.MessageListener;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.SXEController;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.net.SXEMessage;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.records.ElementRecord;
+
+/**
+ * BrowserSXEBridge is the java interface between the SXEController which
+ * contains all model data and the browser which contains the whiteboard and
+ * manipulates the data.
+ */
+public class BrowserSXEBridge {
+
+    private static final Logger LOG = Logger.getLogger(BrowserSXEBridge.class);
+
+    private static final Gson GSON = new Gson();
+
+    private static Gson GSONonlyExposed = new GsonBuilder()
+        .excludeFieldsWithoutExposeAnnotation().create();
+
+    private static final String JAVA_FUNCTION_NAME = "__java_triggerAction";
+
+    private final SXEController sxe;
+
+    private final IWhiteboardBrowser browser;
+
+    private final BrowserActionExecuter executer;
+
+    /**
+     * Creates the action executor and initialises the message listener and the
+     * browser function
+     *
+     * @param sxe
+     *            SXEController created in {@link WhiteboardManager}
+     * @param browser
+     *            the browser created in the view
+     */
+    public BrowserSXEBridge(SXEController sxe, IWhiteboardBrowser browser) {
+        this.sxe = sxe;
+        this.browser = browser;
+        executer = new BrowserActionExecuter(sxe);
+        sxe.addMessageListener(messageListener);
+        browser.createBrowserFunction(JAVA_FUNCTION_NAME, browserFunction);
+        LOG.debug("BrowserSXEBridge has been initialised");
+        // send current state to browser
+        browser.asyncRun("setState("
+            + GSONonlyExposed.toJson(sxe.getDocumentRecord().getRoot()) + ");");
+    }
+
+    private final MessageListener messageListener = new MessageListener() {
+        @Override
+        public void sxeRecordMessageApplied(SXEMessage message) {
+            browser.asyncRun("message(" + GSON.toJson(message) + ");");
+        }
+
+        @Override
+        public void sxeStateMessageApplied(SXEMessage message,
+            ElementRecord root) {
+            browser.asyncRun("setState(" + GSONonlyExposed.toJson(root) + ");");
+        }
+
+        @Override
+        public void sxeMessageSent(SXEMessage message) {
+            // in theory we don't need to do anything here, its however possible
+            // to listen to sent messages so we can also show changes from
+            // the old whiteboard
+            // browser.asyncRun("message(" + GSON.toJson(message) + ");");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=250&issues=AWY_InU7nzfF1D4JrUta&open=AWY_InU7nzfF1D4JrUta)

[ISSUE_KEY]: AWY_InU7nzfF1D4JrUta